src/main/kotlin/com/demonwav/mcdev/i18n/I18nAnnotator.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.intentions.RemoveDuplicatesIntention
+import com.demonwav.mcdev.i18n.intentions.RemoveUnmatchedPropertyIntention
+import com.demonwav.mcdev.i18n.intentions.TrimKeyIntention
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nProperty
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+
+class I18nAnnotator : Annotator {
+    override fun annotate(psiElement: PsiElement, annotationHolder: AnnotationHolder) {
+        if (psiElement is I18nProperty) {
+            checkPropertyMatchesDefault(psiElement, annotationHolder)
+            checkPropertyKey(psiElement, annotationHolder)
+            checkPropertyDuplicates(psiElement, psiElement.parent.children, annotationHolder)
+        }
+    }
+
+    private fun checkPropertyKey(property: I18nProperty, annotations: AnnotationHolder) {
+        if (property.key != property.key.trim { it <= ' ' }) {
```

Comment:
Why do you need a custom predicate for the chars to trim? The regular `.trim()` method already handles various types of white space.